uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/WorkPatternTest.php

```diff
@@ -70,44 +84,74 @@ public function testGetCalendarUsesTheWorkPatternCalendarService() {
     ]);
 
     WorkPatternFabricator::fabricateWithA40HourWorkWeek(['is_default' => true]);
-    $multipleWeekWorkPattern = WorkPatternFabricator::fabricateWithTwoWeeksAnd31AndHalfHours();
 
-    // contract covers the whole period
+    // contracts covering the whole period
     HRJobContractFabricator::fabricate(
-      ['contact_id' => $contact['id']],
+      ['contact_id' => $contact1['id']],
       [
         'period_start_date' => '2015-01-05',
         'period_end_date' => '2015-01-19'
       ]
     );
 
-    // work pattern is effective only after the contract end
-    ContactWorkPatternFabricator::fabricate([
-      'contact_id' => $contact['id'],
-      'pattern_id' => $multipleWeekWorkPattern->id,
-      'effective_date' => CRM_Utils_Date::processDate('2015-02-05'),
-    ]);
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contact2['id']],
+      [
+        'period_start_date' => '2015-01-05',
+        'period_end_date' => '2015-01-19'
+      ]
+    );
 
     $workDayTypes = $this->getWorkDayTypeOptionsArray();
 
-    $calendarDates = civicrm_api3('WorkPattern', 'getCalendar', [
+    $calendars = civicrm_api3('WorkPattern', 'getCalendar', [
       'period_id' => $absencePeriod->id,
-      'contact_id' => $contact['id']
+      'contact_id' => ['IN' => [$contact1['id'], $contact2['id']]]
     ])['values'];
 
+    $this->assertCount(2, $calendars);
+
+    // Since there's not guarantee about the order on which the calendars will
+    // be returned, we get all the contacts ID's from the result, put them in an
+    // array and then check if this array only contains the IDs we're expecting
+    $returnedContactsIDs = array_column($calendars, 'contact_id');
```

Comment:
We can actually sort this array and then
`$this->assertEquals($returnedContactsIDs, [$contact1['id'], $contact2['id']])'